online-shop/src/app/add-product/add-product.component.ts

```diff
@@ -0,0 +1,49 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+import { MatDialogRef } from '@angular/material/dialog';
+import { Supplier } from 'src/supplier';
+import { ProductCategory } from 'src/product-category';
+import { Subscription } from 'rxjs';
+import { ProductCategoryService } from '../services/product-category.service';
+import { SupplierService } from '../services/supplier.service';
+
+@Component({
+    selector: 'app-add-product',
+    templateUrl: './add-product.component.html',
+    styleUrls: ['./add-product.component.scss']
+})
+export class AddProductComponent implements OnInit {
+    addProductForm!: FormGroup;
+    suppliers: Supplier[] = [];
+    productCategories: ProductCategory[] = [];
+    subscription = new Subscription();
+    constructor(
+        private fb: FormBuilder,
```

Comment:
avoid using abbreviations fb, nr, etc... the rule of thumb is the service you inject is called FormBuilder, you should have formBuilder, same applies for categoryService, the class name is ProductCategoryService, if the application grows, chances are that you will have a CategoryService, and a big confusion will start.